src/axom/inlet/Table.cpp

```diff
@@ -295,6 +294,62 @@ std::shared_ptr<Field> Table::addStringHelper(const std::string& name,
   return addField(sidreGroup, axom::sidre::DataTypeId::CHAR8_STR_ID, fullName, name);
 }
 
+bool Proxy::contains(const std::string& name)
+{
+  if(m_table == nullptr)
+  {
+    throw std::out_of_range(
+      "[Inlet] Cannot index a proxy that refers to a field");
+  }
+  return m_table->contains(name);
+}
+
+Proxy Table::operator[](const std::string& name)
+{
+  auto has_table = hasTable(name);
+  auto has_field = hasField(name);
+
+  // Ambiguous case - both a table and field exist with the same name
+  if(has_table && has_field)
+  {
+    std::string msg = fmt::format(
```

Comment:
fwiw, sidre has logic to prevent a child `group` and a `view` within the same group from having the same name. Not sure if that comes into play here.